ocean_lib/common/ddo/credentials.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+from ocean_lib.common.agreements.consumable import ConsumableCodes, MalformedCredential
+
+
+class AddressCredential:
+    def __init__(self, asset):
+        self.asset = asset
+
+    def get_addresses_of_class(self, access_class="allow"):
```

Comment:
`get_addresses_of_class` and `get_lc_addressess_from_entry` return the same thing - the list of allowed or denied addresses (lower case). I think that only `get_addresses_of_class` should exist and it should look like this:

```python
    def get_addresses_of_class(self, access_class="allow"):
        """Get a filtered list of addresses from credentials (use with allow/deny)."""
        address_entry = self.get_address_entry_of_class(access_class)
        return [addr.lower() for addr in address_entry["values"]] if address_entry else []
```